src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexNode.cs

```diff
@@ -1453,13 +1467,15 @@ private RegexNode ReduceConcatenation()
                     return Child(0);
             }
 
-            // Coalesce adjacent characters/strings.
-            ReduceConcatenationWithAdjacentStrings();
-
             // Coalesce adjacent loops.  This helps to minimize work done by the interpreter, minimize code gen,
             // and also help to reduce catastrophic backtracking.
             ReduceConcatenationWithAdjacentLoops();
 
+            // Coalesce adjacent characters/strings.  This is done after the adjacent loop coalescing so that
+            // a One adjacent to both a Multi and a Loop prefers being folded into the Loop rather than into
+            // the Multi.  Doing so helps with auto-atomicity when it's later applied.
```

Comment:
NIT: Don't reset CI for this, but it would be nice if we also add to these comments a concrete example, like: `a+ab` => `(?>a{2,})b`

Reply:
There are examples in the XML comments on the methods being called.